### ISSUE TITLE
An empty array as an initial value makes everything break

### ISSUE THREAD

**Issue by ProofOfKeags** [ianstormtaylor/slate#5351]
**Description**
If you use `value={[]}` which is the sensible thing to do when rendering an empty editor with placeholder text, console is very noisy with all sorts of errors, and basic editor functionality doesn't work.

**Recording**
A GIF or video showing the issue happening. (If you don't include this, there's a very good chance your issue will be closed, because it's much too hard to figure out exactly what is going wrong, and it makes maintenance much harder.)

**Sandbox**
https://codesandbox.io/s/slate-reproductions-forked-ktyjqk

**Steps**
To reproduce the behavior:
1. Follow the directions in the tutorial [here](https://docs.slatejs.org/walkthroughs/01-installing-slate)
2. Omit the last step where you change the initial value
3. Open the console

**Expectation**
I should be able to render an empty editor and not be required to have content in it at the outset

**Environment**
- 0.91.4
- Linux
- Chrome
- TypeScript Version: [e.g. 3.9.7 - required only if it's a TypeScript issue]

**Context**
According to the type definitions, the value field should take anything of type Descendant[] which should include an empty array.

**Comment by dylans** [comment]
It would be an unusable editor as there would be nowhere to focus the cursor.

Yes, we could read an empty value as one that should add an empty text element by default, but then you'd immediately have a different value in the editor than the initialization value. I'm not sure what is better/worse? 🤷‍♂️

**Comment by ProofOfKeags** [comment]
I have also tried initializing with an empty text element and get similar (although not identical) chaotic behavior. If it is a requirement that the editor is initialized with at least one Descendant, this should be documented in the tutorial. The tutorial walks you through doing it but does not state that it is a misuse of the library not to do so.

I'm somewhat surprised this hasn't been encountered before because I figured it'd be a common use case for people to want to initalize editors with no content and just the placeholder text (which doesn't kick in unless the editor is empty).

Am I completely missing how to use this library?

**Comment by drazik** [comment]
I agree that there is a lack of information in the documentation on that point.

The tutorial guides you to add an initial value containing some nodes. And the API shows that the `value` prop is a `Descendant[]` and is required. But I think it should warn the user that the array passed to `value` should not be empty.

I also think it should also be reflected in types by replacing `Descendant[]` by `[Descendant, ...Descendant[]]`. It would make TypeScript user get an error in their editor when passing an empty array to the `value` prop, instead of TypeScript telling them everything is OK and then getting an error at runtime.

What do you think about it?

**Comment by ProofOfKeags** [comment]
> What do you think about it?

The fact that I couldn't initialize an empty editor state caused me to abandon slate in favor of lexical. The recommendations you present above are a definite improvement but ultimately wouldn't have caused me to stay since what I need is to be able to initialize an "empty" state.
